storybookjs/react-native
# feat: csf support [6.0]

Issue: #106 

## What I did

Started working on adding in more of the csf functionality.

The configure function of the preview class is now the result of calling the loadcsf function from core client. This is similar to how the react storybook "start.js" works.

I've renamed the example with CSF in it so its easier to find.

Now configure if called this way and if you use the auto import stories it will generate the code for this.

`configure("react-native", getCSFStories, module, true)`

getCSFStories is a function that returns the require("filepath") calls for csf files.

Fast reload needs to be well tested to confirm that everything works as expected.
--edit 1
So it turns out fast reload is pretty much completely broken 😅

--edit 2
Managed to get fast reload working, it seems like module.hot.dispose data param is not supported in native so I went with a global variable. Not ideal but its at least working now. However I did have to edit loadcsf so it would require changes in storybook or we won't be able to re-use the current loadcsf.

currently there is an issue with non csf stories that I need to debug before merging.

-- final edit?
Non csf stories now also working, so it looks like csf and args are mostly done.

## How to test

In progress

You can start the example app in examples/native and see that the stories are rendering.
